kiliautoml/models/_detectron2_semantic_segmentation.py

```diff
@@ -0,0 +1,520 @@
+            model_name_array=["Kili AutoML"] * len(id_json_list),
+            predictions_probability=[100] * len(id_json_list),
+        )
+        return job_predictions
+
+    @staticmethod
+    def get_contours_instance(instances, i: int):
+        pred_masks = instances.pred_masks
+        masks = np.int8(pred_masks.cpu().detach().numpy())
+
+        im = masks[i] * 255
+        im = np.array(im, dtype=np.uint8)
+        idx = cv2.findContours(im, cv2.RETR_TREE, cv2.CHAIN_APPROX_NONE)
+        list_x_y = idx[0][0]
+        return list_x_y.reshape(-1, 2)
+
+    def get_annotations_from_instances(self, instances, class_names: List[str]):
+        """instances contains multiples bbox and object corresponding to one image"""
+        annotations = []
+        h, w = instances._image_size
+        scores = instances.scores.cpu().detach().numpy()
+        classes = instances.pred_classes.cpu().detach().numpy()
+        for class_i in range(len(classes)):
+            score = scores[class_i]
+            classe = classes[class_i]
+            categories = [Category(name=class_names[classe], confidence=int(score * 100))]
+            list_x_y = self.get_contours_instance(instances, class_i)
+            boundingPoly = [
+                NormalizedVertices(
+                    normalizedVertices=[
+                        NormalizedVertice(x=float(round(x / w, 4)), y=float(round(y / h, 4)))
+                        for x, y in list_x_y
+                    ]
+                )
+            ]
+            annotation = Annotation(
+                boundingPoly=boundingPoly,
+                mid=None,  # type:ignore
+                type="semantic",
+                categories=categories,
+            )
+            annotations.append(annotation)
+        return annotations
+
+    def visualize_predictions(self, visualization_dir, d, dataset_metadata_train, im, outputs):
+        v = Visualizer(
+            im[:, :, ::-1],
+            metadata=dataset_metadata_train,
+            scale=0.5,
+            instance_mode=ColorMode.IMAGE_BW,
+        )
+        out = v.draw_instance_predictions(outputs["instances"].to("cpu"))
+        image_with_predictions = out.get_image()[:, :, ::-1]
+
+        self.save_predictions(visualization_dir, d["file_name"], image_with_predictions)
+
+    def save_predictions(self, visualization_dir, file_name, image_with_predictions):
+        im = Image.fromarray(image_with_predictions)
+        path = os.path.join(visualization_dir, file_name)
+        im.save(path)
+        kili_print("predictions image have been saved in", path)
+
+    def find_errors(
+        self,
+        *,
+        assets: List[AssetT],
+        label_merge_strategy: LabelMergeStrategyT,
+        cv_n_folds: int,
+        epochs: int,
+        batch_size: int,
+        verbose: int = 0,
+        clear_dataset_cache: bool = False,
+        api_key: str = "",
+    ) -> Any:
+        pass
+
+
+# ## DETECTRON FORMAT
+
+
+class ImageCoco(TypedDict):
```

Comment:
the following code could go into a utils/detectron2 folder

Reply:
It could also be directly incorporated to the sdk